core/sql/cli/memorymonitor.cpp

```diff
@@ -223,12 +229,12 @@ void MemoryMonitor::update(float &scale) {
 	char * currPtr;
         bytesRead = fread(buffer, 1, 2048, fd_meminfo_);
         // Make sure there wasn't an error (next fseek will clear eof)
-        if (!feof(fd_meminfo_))// Make sure there wasn't an error
-        {
-	        scale = 6;
-		pressure_ = 0;
-		return;
-        }
+        if (ferror(fd_meminfo_))
+           assert(false); 
+        if (feof(fd_meminfo_))
+           clearerr(fd_meminfo_); 
+        else
+           buffer[bytesRead] = '\0';
```

Comment:
fread cpp reference says the following:
Number of objects read successfully, which may be less than count if an error or end-of-file condition occurs. 

It looks like feof is treated like an error condition. 
This code assumed that buffer is always big enough. I can change it to get rid of this assumption